### DISCUSSION TITLE
First time user experience report

### DISCUSSION THREAD

**Discussion by driftregion** [costinet/AURA#13] [General]
Hi Dan, thanks for publishing AURA. This is my experience getting started with it so far.

- Viewed README page on GitHub, `ToolboxStructure.png` has a transparent background which makes visibility poor in dark mode. (Seems to be default on Windows 11?)  This is not an issue in MATLAB which defaults to rendering HTML docs with a white background.

![image](https://github.com/user-attachments/assets/f57180bd-6f4e-4c53-ad14-e09158d6e041)

fixed here:
https://github.com/driftregion/AURA

![image](https://github.com/user-attachments/assets/6d4dc223-32a4-4ad4-9be1-a186b1385156)

- downloaded package from https://github.com/costinet/AURA/releases
- ran `SweepSeriesCapBuck.mlx`. Very cool! However it failed with the following error:

![image](https://github.com/user-attachments/assets/9e911e58-c46b-41b0-b008-b5ca2bfac7e2)

- Observed `SweepSeriesCapBuck.mlx` is warning about LTSpice directory not found.  It is installed on my machine, but at `"C:\Users\$USER\AppData\Local\Programs\ADI\LTspice\LTspice.exe"` rather than the toolbox default.

**Comment by driftregion** [comment]
I tried to build the package and run the tests but encountered issues with hardcoded and inconsistent paths.

The package build script is here: https://github.com/driftregion/AURA/blob/master/packageToolbox.m

I made a couple changes to directory structure following this resource: https://rse.shef.ac.uk/blog/2022-05-05-concise-guide-to-reproducible-matlab/ 
and am now able to build the package. I believe it should be reproducible for others also.

However, `AURAdbtestCases` fails:

```
>> AURAdbtestCases
Index exceeds the number of array elements. Index must not exceed 0.

Error in indexing (line 402)
                 [varargout{1:nargout}] = builtin('subsref',obj,s);

Error in indexing (line 7)
            varargout = {obj.components(s(1).subs{:})};

Error in AURAdbtestCases (line 49)
    fet = tDB(1);
```

**Reply by costinet** [reply]
Thanks for the feedback.  

The LTSpice directory issue fails safe.  The warning just indicates that the devices from default LTspice Libararies (e.g. standard.mos) won't be available until the LTSpice folder is updated.   That setting is saved between versions/invocations in 
`fullfile(userpath, 'SMPSToolbox', 'AURA', 'parsers','NetlistCircuitParser','settings.mat')`
or setting these variables directly with 
`NetlistCircuitParser().LTSpiceFolder = ...;`
`NetlistCircuitParser().LTSpiceExe= ...;`

The errors, on the other hand, look like they stem from the AURAdb transistor database being empty.  Clearly, that isn't handled gracefully. 

Have you run 
`updateToolbox`
?

A while back, due to file sizes, the component libraries were migrated into their own repo [AURAdb](https://github.com/costinet/AURAdb).  updateToolbox both searches for updates to AURA and syncs the latest AURAdb, which will populate the transistor database.  Looks like either AURAdb hasn't been synced, or something went wrong with the interaction between them.  Do you have anything in the directory specified by 
`fullfile(userpath, 'SMPSToolbox', 'AURAdb')`
?

**Reply by driftregion** [reply]
Thanks for this information Dan.
 
running `updateToolbox` populated the transistor database and `SweepSeriesCapBuck.mlx` runs to completion.
